third_party/terraform/resources/resource_logging_sink.go

```diff
@@ -31,6 +33,20 @@ func resourceLoggingSinkSchema() map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
+
+		"bigquery_options": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"use_partitioned_tables": {
+						Type:     schema.TypeBool,
+						Optional: true,
```

Comment:
Since this original PR we've adopted a new stance to avoid blocks that have a single optional field within them. This was because defining the block without any field inside (eg: `bigquery_options = {}`) can have different interpretations depending on the API behavior and it's also a frequent cause of panics in the provider as it is unintuitive behavior to guard against.

For this PR I think it makes sense to mark this field as `Required` since removing the block appears to be falsey in the api.

Reply:
Okay, so `bigquery_options` is optional, but `use_partitioned_tables` inside it is required, right? I'll fix it.

Reply:
✅ https://github.com/GoogleCloudPlatform/magic-modules/pull/2557/commits/dd3b5e15d4611da17388ce101b28f635bc986079